Standardize scss import practices

brad-decker opened the pull request:
1. Rename the base 'index' scss files to be more descriptive
2. separates ui/app component imports into their own files
3. moves imports to their proper new location
4. Alphabetizes imports in root files.

brad-decker committed:
Standardize scss import practices

1. Rename the base 'index' scss files to be more descriptive
2. separates ui/app component imports into their own files
3. moves imports to their proper new location
4. Alphabetizes imports in root files.

tmashuang commented:
Seems that some of the ordering mattered in a couple of places. A couple of the ones I found

<details>
<summary>Account Details Button Width</summary>
<img src='https://user-images.githubusercontent.com/13376180/90456521-015c9f00-e0ae-11ea-827f-b70d3e3b8e1b.png'>

`.hide-token-confirmation__button, .button`  `width` and `padding` over  `.account-details-modal__button` `width` `padding`
</details>

<details>
<summary>Tx List Item tx category color</summary>
<img src='https://user-images.githubusercontent.com/13376180/90457041-52b95e00-e0af-11ea-92b7-c4d865039383.png'>

`.list-item` `color` over `.transaction-list-item--unconfirmed' `color`
</details>

<details>
<summary>Tx Details Speed Up and Cancel font size</summary>
<img src='https://user-images.githubusercontent.com/13376180/90458526-7f6f7480-e0b3-11ea-8fa7-894f1a2a61ec.png'>

 `.hide-token-confirmation__button, .button` `font-size` over `.transaction-list-item-details__header-button` `font-size`
</details>

<details>
<summary>Send Token button on asset list item</summary>
<img src='https://user-images.githubusercontent.com/13376180/90456135-ef2e3100-e0ac-11ea-9bfe-db950c8851cb.png'>

`button.scss` `font-size` over `.asset-list-item__send-token-button` `font-size`
 `.hide-token-confirmation__button, .button` `display` and `width` over `.asset-list-item__send-token-button` `display` and `width`

I think these are from the old design and can be removed?
</details>

<details>
<summary>Buy button width</summary>
<img src='https://user-images.githubusercontent.com/13376180/90457689-1d157480-e0b1-11ea-80ba-810788da2747.png'>

`.hide-token-confirmation__button, .button` `width` over `.eth-overview__button, .token-overview__button` `width`
</details>

Would we just tack an `!important` for the individual cases?

brad-decker commented:
@tmashuang thank you for finding those! I'll go through and review why the import order caused this to break as it likely means we are relying on styles from unrelated files which most likely is causing more heartaches than fixing things.

brad-decker committed:
fix import order issues

brad-decker commented:
@tmashuang - we can avoid `!important;` by making our css selectors more specific, in this case .button and .hide-token-confirmation__button are equally specific they are both just a single class name. It's important to note that the way we are using BEM is actually making it so that our "nested" scss isn't really creating nested selectors.

```scss
.some-class {
  .child {
     font-size: 16px;
  }
}
```

compiles to
```css
.some-class .child {
  font-size: 16px;
}
```

whereas
```scss
.some-class {
  &__child {
    font-size: 16px;
  }
}
```

compiles to
```css
.some-class__child {
  font-size: 16px;
}
```

In some cases here I am adding an additional ampersand to get the nested effect
```scss
.some-class {
 & &__child {
  font-size: 16px;
 }
}
```

compiles to
```css
.some-class .some-class__child {
  font-size: 16px;
}
```
as we would hope.

brad-decker commented:
I added a commit that resolves those specific issues. I'm looking through the extension now to see if i can find anymore. I did end up pulling out the mixins created ad-hoc in the buttons scss file and converted everywhere they were being used to consume the button component instead. These will likely need to be reviewed carefully for breaking changes

tmashuang commented on ui/app/components/ui/page-container/page-container-header/page-container-header.component.js:
```diff
@@ -72,7 +73,7 @@ export default class PageContainerHeader extends Component {
         {
           onClose && headerCloseText
-            ? <div className="page-container__header-close-text" onClick={() => onClose()}>{ headerCloseText }</div>
+            ? <Button type="link" className="page-container__header-close-text" onClick={() => onClose()}>{ headerCloseText }</Button>
```
Seems this changed the Close to move to a new line and to overlaid the Basic and Advanced Gas tabs, which Selenium has a failed on. 

<details>
<summary>Customize Gas Close modal text</summary>
<img src='https://user-images.githubusercontent.com/13376180/90548759-492d0600-e142-11ea-88df-d9d965bf11ad.png'>
<details>

brad-decker replied:
Good find, I fixed this issue for the choose recipients view on the send page, but missed it here. I'll apply the same fix there and get a revision up for you.

brad-decker committed:
fix gas modal button

brad-decker commented:
Ready for another look @tmashuang. Thank you 🙏

brad-decker commented:
As it turns out, not nesting CSS seems to be the philosophy of BEM. I can achieve the same specificity by changing `& &__element` to `&__element .sub-block`. Either or is fine for this work, but I'll be researching BEM and how we might codify our usage of it.

tmashuang commented on ui/app/pages/pages.scss:
```diff
@@ -1,18 +1,20 @@
+@import 'send/send';
+@import 'settings/index';
 @import 'token/index';
 @import 'create-account/index';
```
```suggestion
@import 'create-account/index';
@import 'unlock-page/index;'
```

brad-decker replied:
whoops, rebase error. Thanks

brad-decker replied:
Fixed, and put in the proper order.

tmashuang commented on ui/app/pages/settings/info-tab/info-tab.component.js:
```diff
@@ -19,71 +20,71 @@ export default class InfoTab extends PureComponent {
+          </Button>
         </div>
       </div>
     )
```
Above changes have centered and padded the About page's Links section

<details>
<summary>About page's link section</summary>
<img src='https://user-images.githubusercontent.com/13376180/90593520-fe8ca780-e19c-11ea-9176-7753e3640e03.png'>
</details>

brad-decker replied:
On it.

brad-decker replied:
<details>
<summary>Fixed</summary>

<img src="https://user-images.githubusercontent.com/4448075/90660339-68439a80-e20b-11ea-9ee7-44ddcaeceda0.png" />

</details>

brad-decker committed:
fix about and unlock page

tmashuang reviewed (approved):
LGTM
